### ISSUE TITLE
TypeError, when read segment to buffer

### ISSUE THREAD

**Issue by MarkHe1222** [anarchy9388/chunky_monkey#16]
2019-07-13 14:18:55,707 [  DEBUG] [read_segment_to_buffer (MainThread)] reading segment to buffer. ssId: 10.243.20.18, partitionId: 0ee46af6-fb3b-4f5f-9e96-2a0f938631ce, filename: 0883, offset: 5044344800, count: 11184800
2019-07-13 14:18:55,723 [  DEBUG] [            _log (Thread-36)] starting thread (client mode): 0x9864fdd0L
2019-07-13 14:18:55,723 [  DEBUG] [            _log (Thread-36)] Local version/idstring: SSH-2.0-paramiko_1.18.5
2019-07-13 14:18:55,773 [  DEBUG] [            _log (Thread-36)] Remote version/idstring: SSH-2.0-OpenSSH_7.2
2019-07-13 14:18:55,775 [   INFO] [            _log (Thread-36)] Connected (version 2.0, client OpenSSH_7.2)
2019-07-13 14:18:55,779 [  DEBUG] [            _log (Thread-36)] kex algos:[u'curve25519-sha256@libssh.org', u'ecdh-sha2-nistp256', u'ecdh-sha2-nistp384', u'ecdh-sha2-nistp521', u'diffie-hellman-group-exchange-sha256', u'diffie-hellman-group14-sha1'] server key:[u'ssh-rsa', u'rsa-sha2-512', u'rsa-sha2-256', u'ssh-dss', u'ecdsa-sha2-nistp256', u'ssh-ed25519'] client encrypt:[u'aes128-ctr', u'aes192-ctr', u'aes256-ctr'] server encrypt:[u'aes128-ctr', u'aes192-ctr', u'aes256-ctr'] client mac:[u'hmac-sha2-256', u'hmac-sha2-512'] server mac:[u'hmac-sha2-256', u'hmac-sha2-512'] client compress:[u'none'] server compress:[u'none'] client lang:[u''] server lang:[u''] kex follows?False
2019-07-13 14:18:55,779 [  DEBUG] [            _log (Thread-36)] Kex agreed: diffie-hellman-group14-sha1
2019-07-13 14:18:55,779 [  DEBUG] [            _log (Thread-36)] HostKey agreed: ecdsa-sha2-nistp256
2019-07-13 14:18:55,779 [  DEBUG] [            _log (Thread-36)] Cipher agreed: aes128-ctr
2019-07-13 14:18:55,779 [  DEBUG] [            _log (Thread-36)] MAC agreed: hmac-sha2-256
2019-07-13 14:18:55,779 [  DEBUG] [            _log (Thread-36)] Compression agreed: none
2019-07-13 14:18:56,012 [  DEBUG] [            _log (Thread-36)] kex engine KexGroup14 specified hash_algo <built-in function openssl_sha1>
2019-07-13 14:18:56,012 [  ERROR] [            _log (Thread-36)] Unknown exception: Too many arguments for this mode
2019-07-13 14:18:56,016 [  ERROR] [            _log (Thread-36)] Traceback (most recent call last):
2019-07-13 14:18:56,017 [  ERROR] [            _log (Thread-36)]   File "/usr/lib/python2.7/site-packages/paramiko/transport.py", line 1798, in run
2019-07-13 14:18:56,017 [  ERROR] [            _log (Thread-36)]     self.kex_engine.parse_next(ptype, m)
2019-07-13 14:18:56,017 [  ERROR] [            _log (Thread-36)]   File "/usr/lib/python2.7/site-packages/paramiko/kex_group1.py", line 75, in parse_next
2019-07-13 14:18:56,017 [  ERROR] [            _log (Thread-36)]     return self._parse_kexdh_reply(m)
2019-07-13 14:18:56,017 [  ERROR] [            _log (Thread-36)]   File "/usr/lib/python2.7/site-packages/paramiko/kex_group1.py", line 112, in _parse_kexdh_reply
2019-07-13 14:18:56,017 [  ERROR] [            _log (Thread-36)]     self.transport._activate_outbound()
2019-07-13 14:18:56,017 [  ERROR] [            _log (Thread-36)]   File "/usr/lib/python2.7/site-packages/paramiko/transport.py", line 2135, in _activate_outbound
2019-07-13 14:18:56,017 [  ERROR] [            _log (Thread-36)]     engine = self._get_cipher(self.local_cipher, key_out, IV_out)
2019-07-13 14:18:56,017 [  ERROR] [            _log (Thread-36)]   File "/usr/lib/python2.7/site-packages/paramiko/transport.py", line 1666, in _get_cipher
2019-07-13 14:18:56,017 [  ERROR] [            _log (Thread-36)]     return self._cipher_info[name]['class'].new(key, self._cipher_info[name]['mode'], '', counter)
2019-07-13 14:18:56,017 [  ERROR] [            _log (Thread-36)]   File "/usr/lib64/python2.7/site-packages/Crypto/Cipher/AES.py", line 232, in new
2019-07-13 14:18:56,018 [  ERROR] [            _log (Thread-36)]     return _create_cipher(sys.modules[__name__], key, mode, *args, **kwargs)
2019-07-13 14:18:56,018 [  ERROR] [            _log (Thread-36)]   File "/usr/lib64/python2.7/site-packages/Crypto/Cipher/__init__.py", line 75, in _create_cipher
2019-07-13 14:18:56,018 [  ERROR] [            _log (Thread-36)]     raise TypeError("Too many arguments for this mode")
2019-07-13 14:18:56,018 [  ERROR] [            _log (Thread-36)] TypeError: Too many arguments for this mode
2019-07-13 14:18:56,018 [  ERROR] [            _log (Thread-36)]

**Comment by MarkHe1222** [comment]
admin@provo-platinum:/tmp/test/chunky_monkey> hostname -i
10.243.20.15
admin / ChangMe 

admin@provo-platinum:/tmp/test/chunky_monkey> pwd
/tmp/test/chunky_monkey

`admin@provo-platinum:/tmp/test/chunky_monkey> python chunky_monkey.py -ns dump -cid 00009ef4-aa76-44c6-852c-5f13bd579298 -s 1`

**Comment by anarchy9388** [comment]
```
admin@provo-platinum:/tmp/test/chunky_monkey> python chunky_monkey.py -ns dump -cid 00009ef4-aa76-44c6-852c-5f13bd579298 -s 1
chunky_monkey.py v1.0.dev7 (svc_tools v1.1.5)                 Started 2019-07-16 17:22:11

Output log : /tmp/chunky_monkey/chunky_monkey_2019-07-16-17-22-11.log
Debug log  : /tmp/chunky_monkey/chunky_monkey_2019-07-16-17-22-11.log

1. Getting chunk info for chunkId 00009ef4-aa76-44c6-852c-5f13bd579298..DONE
2. Collecting target segment detail..DONE
   Target segment:
{'checksum': '1376589904',
 'endOffset': '22369600',
 'locationType': 'STORAGE_SERVER',
 'offset': '11184800',
 'sequence': '1',
 'ssLocation': {'filename': '0883',
                'offset': '5044344800',
                'partitionId': '0ee46af6-fb3b-4f5f-9e96-2a0f938631ce',
                'partitionIdUuid': {'highBytes': '1073100220230487903',
                                    'lowBytes': '11427367350887133646'},
                'ssId': 'd474cfda-6e2e-4600-a7e9-583d02333d6a',
                'ssIdUuid': {'highBytes': '15309089570397505024',
                             'lowBytes': '12099298892960709994'},
                'status': 'HEALTHY',
                'statusSn': '-1',
                'tag': '1562524332130'}}
ERROR     Unknown exception: Too many arguments for this mode
ERROR     Traceback (most recent call last):
ERROR       File "/usr/lib/python2.7/site-packages/paramiko/transport.py", line 1798, in run
ERROR         self.kex_engine.parse_next(ptype, m)
ERROR       File "/usr/lib/python2.7/site-packages/paramiko/kex_group1.py", line 75, in parse_next
ERROR         return self._parse_kexdh_reply(m)
ERROR       File "/usr/lib/python2.7/site-packages/paramiko/kex_group1.py", line 112, in _parse_kexdh_reply
ERROR         self.transport._activate_outbound()
ERROR       File "/usr/lib/python2.7/site-packages/paramiko/transport.py", line 2135, in _activate_outbound
ERROR         engine = self._get_cipher(self.local_cipher, key_out, IV_out)
ERROR       File "/usr/lib/python2.7/site-packages/paramiko/transport.py", line 1666, in _get_cipher
ERROR         return self._cipher_info[name]['class'].new(key, self._cipher_info[name]['mode'], '', counter)
ERROR       File "/usr/lib64/python2.7/site-packages/Crypto/Cipher/AES.py", line 232, in new
ERROR         return _create_cipher(sys.modules[__name__], key, mode, *args, **kwargs)
ERROR       File "/usr/lib64/python2.7/site-packages/Crypto/Cipher/__init__.py", line 75, in _create_cipher
ERROR         raise TypeError("Too many arguments for this mode")
ERROR     TypeError: Too many arguments for this mode
ERROR
3. Reading segment data from disk ..Traceback (most recent call last):
  File "chunky_monkey.py", line 1116, in <module>
    chunky_monkey_inst.main()
  File "chunky_monkey.py", line 1031, in main
    self.do_dump(chunk_id=chunk_id, sequence=sequence, is_normal=is_normal)
  File "chunky_monkey.py", line 864, in do_dump
    target_segment['ssLocation']['offset'], int(target_segment['endOffset']) - int(target_segment['offset']))
  File "chunky_monkey.py", line 382, in read_segment_to_buffer
    ssh_client.connect(hostname='{}'.format(ssId), username='admin')
  File "/usr/lib/python2.7/site-packages/paramiko/client.py", line 353, in connect
    t.start_client(timeout=timeout)
  File "/usr/lib/python2.7/site-packages/paramiko/transport.py", line 494, in start_client
    raise e
TypeError: Too many arguments for this mode
admin@provo-platinum:/tmp/test/chunky_monkey>
```

**Comment by anarchy9388** [comment]
@MarkHe1222 

It's a bug in older version of Paramiko, I will fix in new branch :)

```
admin@provo-platinum:/tmp/test/chunky_monkey> sudo python -m pip install --upgrade paramiko
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7.
Collecting paramiko
  Downloading https://files.pythonhosted.org/packages/4b/80/74dace9e48b0ef923633dfb5e48798f58a168e4734bca8ecfaf839ba051a/paramiko-2.6.0-py2.py3-none-any.whl (199kB)
     |################################| 204kB 2.3MB/s
Collecting pynacl>=1.0.1 (from paramiko)
  Downloading https://files.pythonhosted.org/packages/b3/25/e605574f24948a8a53b497744e93f061eb1dbe7c44b6465fc1c172d591aa/PyNaCl-1.3.0-cp27-cp27mu-manylinux1_x86_64.whl (762kB)
     |################################| 768kB 446kB/s
Collecting bcrypt>=3.1.3 (from paramiko)
  Downloading https://files.pythonhosted.org/packages/ad/36/9a0227d048e98409f012570f7bef8a8c2373b9c9c5dfbf82963cbae05ede/bcrypt-3.1.7-cp27-cp27mu-manylinux1_x86_64.whl (59kB)
     |################################| 61kB 2.0MB/s
Collecting cryptography>=2.5 (from paramiko)
  Downloading https://files.pythonhosted.org/packages/e6/68/50698ce24c61db7d44d93a5043c621a0ca7839d4ef9dff913e6ab465fc92/cryptography-2.7-cp27-cp27mu-manylinux1_x86_64.whl (2.3MB)
     |################################| 2.3MB 2.9MB/s
Collecting six (from pynacl>=1.0.1->paramiko)
  Downloading https://files.pythonhosted.org/packages/73/fb/00a976f728d0d1fecfe898238ce23f502a721c0ac0ecfedb80e0d88c64e9/six-1.12.0-py2.py3-none-any.whl
Collecting cffi>=1.4.1 (from pynacl>=1.0.1->paramiko)
  Downloading https://files.pythonhosted.org/packages/8d/e9/0c8afd1579e5cf7bc0f06fbcd7cdb954cbc0baadd505973949a99337da1c/cffi-1.12.3-cp27-cp27mu-manylinux1_x86_64.whl (415kB)
     |################################| 419kB 3.5MB/s
Collecting asn1crypto>=0.21.0 (from cryptography>=2.5->paramiko)
  Downloading https://files.pythonhosted.org/packages/ea/cd/35485615f45f30a510576f1a56d1e0a7ad7bd8ab5ed7cdc600ef7cd06222/asn1crypto-0.24.0-py2.py3-none-any.whl (101kB)
     |################################| 102kB 5.9MB/s
Collecting enum34; python_version < "3" (from cryptography>=2.5->paramiko)
  Downloading https://files.pythonhosted.org/packages/c5/db/e56e6b4bbac7c4a06de1c50de6fe1ef3810018ae11732a50f15f62c7d050/enum34-1.1.6-py2-none-any.whl
Collecting ipaddress; python_version < "3" (from cryptography>=2.5->paramiko)
  Downloading https://files.pythonhosted.org/packages/fc/d0/7fc3a811e011d4b388be48a0e381db8d990042df54aa4ef4599a31d39853/ipaddress-1.0.22-py2.py3-none-any.whl
Collecting pycparser (from cffi>=1.4.1->pynacl>=1.0.1->paramiko)
  Downloading https://files.pythonhosted.org/packages/68/9e/49196946aee219aead1290e00d1e7fdeab8567783e83e1b9ab5585e6206a/pycparser-2.19.tar.gz (158kB)
     |################################| 163kB 6.6MB/s
Installing collected packages: six, pycparser, cffi, pynacl, bcrypt, asn1crypto, enum34, ipaddress, cryptography, paramiko
  Running setup.py install for pycparser ... done
  Found existing installation: paramiko 1.18.5
    Uninstalling paramiko-1.18.5:
      Successfully uninstalled paramiko-1.18.5
Successfully installed asn1crypto-0.24.0 bcrypt-3.1.7 cffi-1.12.3 cryptography-2.7 enum34-1.1.6 ipaddress-1.0.22 paramiko-2.6.0 pycparser-2.19 pynacl-1.3.0 six-1.12.0
admin@provo-platinum:/tmp/test/chunky_monkey> python chunky_monkey.py -ns dump -cid 00009ef4-aa76-44c6-852c-5f13bd579298 -s 1
chunky_monkey.py v1.0.dev7 (svc_tools v1.1.5)                 Started 2019-07-16 17:39:32

Output log : /tmp/chunky_monkey/chunky_monkey_2019-07-16-17-39-32.log
Debug log  : /tmp/chunky_monkey/chunky_monkey_2019-07-16-17-39-32.log

1. Getting chunk info for chunkId 00009ef4-aa76-44c6-852c-5f13bd579298..DONE
2. Collecting target segment detail..DONE
   Target segment:
{'checksum': '1376589904',
 'endOffset': '22369600',
 'locationType': 'STORAGE_SERVER',
 'offset': '11184800',
 'sequence': '1',
 'ssLocation': {'filename': '0883',
                'offset': '5044344800',
                'partitionId': '0ee46af6-fb3b-4f5f-9e96-2a0f938631ce',
                'partitionIdUuid': {'highBytes': '1073100220230487903',
                                    'lowBytes': '11427367350887133646'},
                'ssId': 'd474cfda-6e2e-4600-a7e9-583d02333d6a',
                'ssIdUuid': {'highBytes': '15309089570397505024',
                             'lowBytes': '12099298892960709994'},
                'status': 'HEALTHY',
                'statusSn': '-1',
                'tag': '1562524332130'}}
3. Reading segment data from disk ..DONE
4. Saving segment data to local ..DONE
   CHECKSUM:
       Stored: 1376589904  New-version: 1376589904  Old-version: -1
   File saved in: /tmp/00009ef4-aa76-44c6-852c-5f13bd579298_ec_segment_sequence_1


Execution Time:
3 seconds
admin@provo-platinum:/tmp/test/chunky_monkey>
```

**Comment by anarchy9388** [comment]
@MarkHe1222 

Fixed in issue-16 branch, here's the PR: #17
